### ISSUE TITLE
Wait for owner dialog not closing when the authenticated user joins the meeting

### ISSUE THREAD

**Issue by ashiqhassan95** [jitsi/jitsi-meet#11413]
### Description:

Wait for the owner dialog is not close automatically once the authenticated user joins the meeting. 

![ksnip_20220421-111020](https://user-images.githubusercontent.com/2870845/164382248-e81ee48f-bb87-44eb-a687-7cfa42095576.png)

### Steps to reproduce:

1. User A (Guest) tries to join in a meeting room (room without authenticated users joined already) as a Guest.
2. Now wait for the owner dialog will show for User A.
3. User B (auth user) joins the same meeting room from another window as an authenticated user.
4. User B sees User A in the meeting.
5. User A also enters the meeting room and can see other participants, but **Wait for the owner** dialog is still there.

### Expected behavior:

Wait for Owner dialog should be closed automatically once authenticated user join the room.

### Actual behavior:

Wait for Owner dialog is not closing.

### Server information:

- Jitsi Meet version: 2.0.7210-1
- Operating System: Ubuntu ubuntu 20.04 lts

### Client information:

- Browser / app version: Chrome (100.0.4896.127)
- Operating System: Ubuntu 18.04

### Additional information:

This issue is just gone when uncommenting the code in Line 78 (`store.dispatch(waitForOwner())`)

https://github.com/jitsi/jitsi-meet/blob/dde8c586da37156db67bd8ecbc1e1ecd956c7b2e/react/features/authentication/middleware.web.js#L73-L80

**Comment by saghul** [comment]
Ping @damencho (damn this auth thing!)

**Comment by gpatel-fr** [comment]
I can't repro. Exact same environment - I tested with 2 users as in the description, and 4 as shown in the screenshot.

**Comment by ashiqhassan95** [comment]
> I can't repro. Exact same environment - I tested with 2 users as in the description, and 4 as shown in the screenshot.

Please make sure that you are checking in the jitsi meet environment with a secure domain enabled (based on JWT).

**Comment by gpatel-fr** [comment]
My test environment don't use JWT, it's standard secure domain. The 'anonymous user waits for JWT authenticated user' is not a standard JWT feature I think, from what I remember from the forum I have seen it asked for by people using JWT, so I think it's something you worked out for your needs.

**Comment by damencho** [comment]
> > I can't repro. Exact same environment - I tested with 2 users as in the description, and 4 as shown in the screenshot.
> 
> Please make sure that you are checking in the jitsi meet environment with a secure domain enabled (based on JWT).

You are not suppose to see this dialog when you use jwt. Waiting for moderator is only for secure domain.

**Comment by ashiqhassan95** [comment]
> > > I can't repro. Exact same environment - I tested with 2 users as in the description, and 4 as shown in the screenshot.
> > 
> > 
> > Please make sure that you are checking in the jitsi meet environment with a secure domain enabled (based on JWT).
> 
> You are not suppose to see this dialog when you use jwt. Waiting for moderator is only for secure domain.

My jitsi meet environment based on secure domain (as specified in https://jitsi.github.io/handbook/docs/devops-guide/secure-domain/#jicofo-configuration). In my jitsi meet instance authenticated users enters the meeting room using JWT not by prosody XMPP user credential, participants who joining meeting without JWT is guest user.

If any guest user tries to join a meeting room which has no users (auth) in it, then they have to wait for the authenticated user to arrive in order to join the meeting, until then **Wait for owner** dialog will be displayed for them. 

Once a authenticated user join the room, all the waiting user (guest) will be automatically enter the room and that **Wait for owner** dialog disappear. It was working properly till the latest stable version.

But in the new version, the **Wait for owner** dialog still appear there and other things works fine (can hear and see participants each other like in the screenshot).

**Comment by damencho** [comment]
Yep, this is the mixture of jwt and secure domain what I was talking about, this is case which we do not test when doing changes. Any PRs with fixes are welcome. 
You can try bisect it and find the problem commit.

**Comment by gpatel-fr** [comment]
My money is on [this one](https://github.com/jitsi/jitsi-meet/pull/11074)

**Comment by saghul** [comment]
Unlikely since that only changed code used on the mobile apps.

I can take a look at this next week. Ping me if I don't report back :-)

**Comment by gpatel-fr** [comment]
@saghul 
Yes I have changed my mind about that, see[ #11432](https://github.com/jitsi/jitsi-meet/issues/11432). And I can repro too. Don't understand why I could not before. I must have done a stupid mistake while testing.

**Comment by saghul** [comment]
Fix in https://github.com/jitsi/jitsi-meet/pull/11463
